packages/javascript/bh-shared-ui/src/views/SSOConfiguration/SSOConfiguration.tsx

```diff
@@ -0,0 +1,281 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState, FC, useCallback, useMemo, ChangeEvent } from 'react';
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Box, Grid, Typography, Paper, TextField, useTheme } from '@mui/material';
+import { useMutation, useQuery } from 'react-query';
+import { CreateOIDCProvideRequest, SSOProvider } from 'js-client-library';
+import { apiClient } from '../../utils';
+import {
+    CreateSAMLProviderDialog,
+    CreateSAMLProviderFormInputs,
+    CreateMenu,
+    ConfirmationDialog,
+    DocumentationLinks,
+    SSOProviderTable,
+    PageWithTitle,
+    SSOProviderInfoPanel,
+} from '../../components';
+import CreateOIDCProviderDialog from '../../components/CreateOIDCProviderDialog';
+import { useFeatureFlag } from '../../hooks';
+import { useNotifications } from '../../providers';
+
+const SSOConfiguration: FC = () => {
+    /* Hooks */
+    const theme = useTheme();
+    const { addNotification } = useNotifications();
+    const { data: flag } = useFeatureFlag('oidc_support');
+
+    const [selectedSSOProviderId, setSelectedSSOProviderId] = useState<SSOProvider['id'] | undefined>();
+    const [ssoProviderIdToDelete, setSSOProviderIdToDelete] = useState<SSOProvider['id'] | undefined>();
+    const [dialogOpen, setDialogOpen] = useState<'SAML' | 'OIDC' | 'DELETE' | ''>('');
+    const [nameFilter, setNameFilter] = useState<string>('');
+    const [createProviderError, setCreateProviderError] = useState<string>('');
+    const [typeSortOrder, setTypeSortOrder] = useState<'asc' | 'desc' | undefined>();
+
+    const listSSOProvidersQuery = useQuery(['listSSOProviders'], ({ signal }) =>
+        apiClient.listSSOProviders({ signal }).then((res) => res.data.data)
```

Comment:
Recap from offline conversation:
- This API is feature flagged
- Two options:
  - massage data from existing list saml providers endpoint into sso provider format
  - apply saml only filter to sso-providers endpoint when oidc feature flag is disabled

Reply:
@iustinum is on it, we'll fix this via API side of things